jax/core.py

```diff
@@ -311,18 +298,19 @@ def full_raise(self, val):
       elif val._trace.sublevel < sublevel:
         return self.sublift(val)
       else:
-        escaped_tracer_error(
-          "Can't lift sublevels {} to {}".format(val._trace.sublevel, sublevel))
+        raise escaped_tracer_error("Can't lift sublevels {} to {}"
```

Comment:
`escaped_tracer_error` is a function, not an `Exception`, so this line isn't going to work

Reply:
It's a function that returns an exception. I think the tests cover it and it works. (I switched this to make mypy happy.)

Reply:
```python
In [1]: def foo():
   ...:     return Exception("shoyer")
   ...:

In [2]: raise foo()
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-2-b2ecd83f935c> in <module>
----> 1 raise foo()

Exception: shoyer
```

Reply:
Thanks for reviewing all the code btw! I feel safer knowing the eagle eye of shoyer is ever-watchful.

Reply:
Oh, that is subtle. I'm glad we have mypy for checks like this :)

Reply:
Just for posterity: the mypy issue I ran into was that it complained there were missing return statements on this function (since the type signature I added was to return a non-null value). It didn't know that the function we were calling would just raise an exception. There may be other solutions, but this was the one that sprang to mind!